core-bundle/src/Cron/Cron.php

```diff
@@ -120,13 +124,33 @@ public function run(string $scope): void
             $repository->unlockTable();
         }
 
+        $exception = null;
+
         // Execute all cron jobs to be run
         foreach ($cronJobsToBeRun as $cron) {
-            if (null !== $this->logger) {
-                $this->logger->debug(sprintf('Executing cron job "%s"', $cron->getName()));
+            try {
+                if (null !== $this->logger) {
+                    $this->logger->debug(sprintf('Executing cron job "%s"', $cron->getName()));
+                }
+
+                $cron($scope);
+            } catch (CronExecutionSkippedException $e) {
+                // Restore previous run date in case cronjob skips itself
+                /** @var CronJobEntity $lastRunEntity */
+                $lastRunEntity = $repository->findOneByName($cron->getName());
+                $lastRunEntity->setLastRun($cron->getPreviousRun());
+                $entityManager->flush();
+            } catch (\Throwable $e) {
+                // Catch any exceptions so that other cronjobs are still executed
```

Comment:
Should we add something like this here?
```php
if (null !== $this->logger) {
    $this->logger->error((string) $e);
}
```

Reply:
Yes, but not for the first one, since it will be thrown (and thus logged), right?

Reply:
I’d probably do it for all of them to make the output more consistent and less confusing